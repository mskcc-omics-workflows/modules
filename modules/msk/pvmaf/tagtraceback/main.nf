process PVMAF_TAGTRACEBACK {
    tag "$meta.id"
    label 'process_single'
    //ghcr.io/msk-access/postprocessing_variant_calls:0.2.8
    // TODO nf-core: List required Conda package(s).
    //               Software MUST be pinned to channel (i.e. "bioconda"), version (i.e. "1.10").
    //               For Conda, the build (i.e. "h9402c20_2") must be EXCLUDED to support installation on different operating systems.
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'ghcr.io/msk-access/postprocessing_variant_calls:type_traceback_0.0.7':
        'ghcr.io/msk-access/postprocessing_variant_calls:type_traceback_0.0.7' }"

    input:
    tuple val(meta), path(maf)
    path(sample_sheets)

    output:
    tuple val(meta), path("*.maf"), emit: maf
    path "versions.yml"           , emit: versions
    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix != null ? "${task.ext.prefix}" : (meta.patient != null ? "${meta.patient}" : "")
    def sampleFiles = sample_sheets ? sample_sheets.collect { file -> "-sheet $file" }.join(' ') : ''
    def output = prefix ? "${prefix}_traceback.maf": "multi_sample_traceback.maf"

    """
    pv maf tag traceback \\
    -m $maf \\
    $sampleFiles \\
    --output $output \\
    $args


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pv: \$( pv --version  )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix != null ? "${task.ext.prefix}" : (meta.patient != null ? "${meta.patient}" : "")
    def output = prefix ? "${prefix}_traceback.maf": "multi_sample_traceback.maf"
    """
    touch $output
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pv: \$( pv --version  )
    END_VERSIONS
    """
}
