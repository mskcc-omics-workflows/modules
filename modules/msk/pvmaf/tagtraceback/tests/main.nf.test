nextflow_process {

    name "Test Process PVMAF_TAGTRACEBACK"
    script "../main.nf"
    process "PVMAF_TAGTRACEBACK"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "pvmaf"
    tag "pvmaf/tagtraceback"

    test("chr22mafstandard - msk - traceback") {


        when {
            process {
                """input[0] = [
                            [ id:'chr22', patient:'test'], // meta map
                            file(params.test_data_mskcc['chr22']['chr22mafstandard'], checkIfExists: true),
                        ]
                    input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("chr22mafstandard - msk - traceback - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                            [ id:'chr22', patient:'test'], file(params.test_data_mskcc['chr22']['chr22mafstandard'], checkIfExists: true),
                        ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
