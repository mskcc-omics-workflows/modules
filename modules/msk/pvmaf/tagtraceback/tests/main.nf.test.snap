{
    "chr22mafstandard - msk - traceback": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_traceback.maf:md5,23ca5bc6918acfd84b98fdce78bc008b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,aabbb21415be8383794da07e3b15b062"
                ],
                "maf": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_traceback.maf:md5,23ca5bc6918acfd84b98fdce78bc008b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aabbb21415be8383794da07e3b15b062"
                ]
            }
        ],
        "timestamp": "2024-07-09T10:15:08.586507"
    },
    "chr22mafstandard - msk - traceback - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_traceback.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,aabbb21415be8383794da07e3b15b062"
                ],
                "maf": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_traceback.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aabbb21415be8383794da07e3b15b062"
                ]
            }
        ],
        "timestamp": "2024-07-09T10:16:09.337412"
    }
}